src/server/ua_server.c

```diff
@@ -392,8 +392,8 @@ UA_Server_newWithConfig(UA_ServerConfig *config) {
     if(server->config.eventLoop->logger == &config->logger)
         server->config.eventLoop->logger = &server->config.logger;
 
-    if(server->config.certificateVerification.logger == &config->logger)
-        server->config.certificateVerification.logger = &server->config.logger;
+    /* re-set the logger pointer */
+    server->config.pLogger = &server->config.logger;
```

Comment:
Only reset the pointer if
- pLogger is NULL
- config->pLogger == &config->logger (for the original config)

That way it is already possible to use the pLogger without the "old" logger.

Reply:
Added check